docs/events.md

```diff
@@ -0,0 +1,137 @@
+
+#### ObjectUpdatedEvent
+- **Class**: `OCA\OpenRegister\Event\ObjectUpdatedEvent`
+- **Triggered**: When an existing object is updated in a register
+- **Data Provided**:
+  - `getObject()`: Returns the updated ObjectEntity
+  - `getOldObject()`: Returns the ObjectEntity before updates
+- **Usage**: Useful for tracking changes to objects, auditing modifications, or triggering follow-up actions
+
+#### ObjectDeletedEvent
+- **Class**: `OCA\OpenRegister\Event\ObjectDeletedEvent`
+- **Triggered**: When an object is deleted from a register
+- **Data Provided**:
+  - `getObject()`: Returns the ObjectEntity that was deleted
+- **Usage**: Can be used for cleanup operations, maintaining related data integrity, or sending notifications about deletions
+
+## Using Events
+
+Events are a powerful way to decouple different parts of your application and respond to changes in the system. The OpenRegister app uses Nextcloud's event dispatcher system to broadcast various events that you can listen to.
+
+### Event Handling Overview
+
+Events are dispatched at key points in the application lifecycle, such as when objects are created, updated, or deleted. By implementing event listeners, you can:
+
+- Perform additional actions when changes occur
+- Maintain data consistency across different parts of the system 
+- Send notifications or trigger external integrations
+- Add custom business logic without modifying core code
+- Create audit trails and logs
+
+### Implementation Steps
+
+To start handling events in your application, follow these steps:
+
+### 1. Create an Event Listener Class
+
+First, create a class that will handle the event. Your listener class should:
+
+- Implement the `IEventListener` interface
+- Define a `handle()` method that receives the event object
+- Be placed in an appropriate namespace in your application
+
+Here's a basic example structure:
+
+```php
+namespace OCA\MyApp\Listener;
+
+use OCA\OpenRegister\Event\SchemaCreatedEvent;
+use OCA\OpenRegister\Event\SchemaUpdatedEvent;
+use OCA\OpenRegister\Event\SchemaDeletedEvent;
+
+class SchemaListener implements IEventListener {
+    public function handle(SchemaCreatedEvent $event) {
+        // Handle the event
+    }
+}
+```
+
+### 2. Register the Event Listener
+
+After creating your listener class, you need to register it with Nextcloud's event dispatcher system. This is done in your application's `lib/AppInfo/Application.php` file by registering the listener in the `register()` method:
+
+```php
+/* @var IEventDispatcher $dispatcher */
+$dispatcher = $this->getContainer()->get(IEventDispatcher::class);
+$dispatcher->addServiceListener(eventName: SchemaCreatedEvent::class, className: SchemaCreatedEvent::class);
+```
+
+This line registers the `SchemaListener` class to handle `SchemaCreatedEvent` events. You can add similar lines for other events to register your listener for those specific events.
```

Comment:
Minor Grammar: In the sentence "You can add similar lines for other events to register your listener for those specific events," consider rephrasing for better clarity:
"You can register listeners for other events by adding similar lines."